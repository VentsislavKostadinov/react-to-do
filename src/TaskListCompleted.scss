@import './font.scss';
@import './style/variables';
@import './style/mixins';

.completed-list {
    @include task-styling;

    .list-group-item {
        border: none;
        background: inherit;
        margin: 0.5rem;
        background: $background-task;
        width: 100%;
        color: $deleted-task;

        &:hover {
            background: $hover-task;
        }
    }
}
