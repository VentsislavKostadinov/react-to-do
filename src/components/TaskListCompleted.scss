@import '../style/font.scss';
@import '../style/variables';
@import '../style/mixins';

.task-list-wrap {
    @include task-styling;
     background-color: inherit;
     hr {
        color: white !important;
     }
    .list-group-item {
        border: none;
        background: inherit; 
        margin: 0.5rem;
        width: 100%;
        color: $deleted-task;

        &:hover {
            background: $hover-task;
        }
    }
}
