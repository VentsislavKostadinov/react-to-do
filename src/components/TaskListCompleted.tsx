import React from 'react'
import { TaskListCompletedProps } from '../model/TaskListCompletedProps.types'
import { Col, Container, Row } from 'react-bootstrap'
import classes from '../style/TaskListCompleted.module.scss'
import { CommonHeadline } from '../common/CommonHeadline'
import { TaskProps } from '../model/TaskProps.types'

export const TaskListCompleted = ({ tasks }: TaskListCompletedProps) => {
    return (
        <Container>
            <Row>
                <Col>
                    {tasks.length !== 0 ? (
                        <>
                            <CommonHeadline
                                dataTestid="completed-tasks"
                                title="Completed Tasks"
                                heading="h2"
                            />
                            <div className={classes.taskCompleteWrap}>
                                {tasks.map(
                                    (task: TaskProps | any, index: number) =>
                                        task.completed && (
                                            <div key={index}>
                                                <div
                                                    className={
                                                        classes.completedItems
                                                    }
                                                    key={task.id}
                                                >
                                                    <del>
                                                        {task.description}
                                                    </del>
                                                </div>
                                                <hr />
                                            </div>
                                        ),
                                )}
                            </div>
                        </>
                    ) : null}
                </Col>
            </Row>
        </Container>
    )
}
