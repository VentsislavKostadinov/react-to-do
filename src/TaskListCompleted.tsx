import React from 'react'
import { TaskListCompletedProps } from './model/TaskListCompletedProps.types'
import { Container, ListGroup } from 'react-bootstrap'
import './TaskListCompleted.scss'
import { CommonHeadline } from './common/CommonHeadline'
import { TaskProps } from './model/TaskProps.types'

export const TaskCompletedList = ({ tasks }: TaskListCompletedProps) => {
    return (
        <Container>
            {tasks.length !== 0 ? (
                <>
                    <CommonHeadline title="Completed Tasks" heading="h2" />
                    <ListGroup className="completed-list">
                        {tasks.map(
                            (task: TaskProps | any) =>
                                task.completed && (
                                    <ListGroup.Item key={task.id}>
                                        <del>{task.description}</del>
                                    </ListGroup.Item>
                                ),
                        )}
                    </ListGroup>
                </>
            ) : null}
        </Container>
    )
}
