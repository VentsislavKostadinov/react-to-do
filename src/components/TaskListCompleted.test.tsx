import React from 'react'
import { TaskListCompleted } from './TaskListCompleted'
import { TaskListCompletedProps } from '../model/TaskListCompletedProps.types'
import { render, screen } from '@testing-library/react'

describe('TaskListCompleted component', () => {
    const taskListCompletedProps: TaskListCompletedProps = {
        tasks: [
            {
                task: {
                    id: 4534534,
                    description: 'text description',
                    completed: false,
                },
                onEdit: jest.fn,
                onComplete: jest.fn,
                onDelete: jest.fn,
            },
        ],
    }

    it('renders correctly', () => {
        const { asFragment } = render(
            <TaskListCompleted {...taskListCompletedProps} />,
        )
        expect(asFragment).toMatchSnapshot()
    })

    it('displayed correctly', () => {
        render(<TaskListCompleted {...taskListCompletedProps} />)

        const completedTasksTitle = screen.getByTestId('completed-tasks')
        expect(completedTasksTitle).toBeInTheDocument

        const container = document.querySelector('.container')
        expect(container?.innerHTML).toContain('Completed Tasks')
    })
})
