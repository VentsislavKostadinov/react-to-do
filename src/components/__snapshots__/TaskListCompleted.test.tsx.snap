// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TaskListCompleted component renders correctly 1`] = `[Function]`;
